state/txindex/indexer_service.go

```diff
@@ -54,20 +61,31 @@ func (is *IndexerService) OnStart() error {
 			eventDataHeader := msg.Data().(types.EventDataNewBlockHeader)
 			height := eventDataHeader.Header.Height
 			batch := NewBatch(eventDataHeader.NumTxs)
+
 			for i := int64(0); i < eventDataHeader.NumTxs; i++ {
 				msg2 := <-txsSub.Out()
 				txResult := msg2.Data().(types.EventDataTx).TxResult
+
 				if err = batch.Add(&txResult); err != nil {
-					is.Logger.Error("Can't add tx to batch",
+					is.Logger.Error(
+						"failed to add tx to batch",
 						"height", height,
 						"index", txResult.Index,
-						"err", err)
+						"err", err,
+					)
 				}
 			}
-			if err = is.idr.AddBatch(batch); err != nil {
-				is.Logger.Error("Failed to index block", "height", height, "err", err)
+
+			if err := is.blockIdxr.Index(eventDataHeader); err != nil {
+				is.Logger.Error("failed to index block", "height", height, "err", err)
+			} else {
+				is.Logger.Error("indexed block", "height", height, "err", err)
+			}
```

Comment:
Here we piggy-back off of the existing indexer.

Reply:
should we return in these cases? or collect the errors in some way?